File: TransCoder_tokenized_test_set_functions/cpp/COUNT_ARRAYS_ADJACENT_ELEMENTS_ONE_DIVIDE_ANOTHER.cpp
int numofArray ( int n, int m ) {
  int dp [ MAX ] [ MAX ];
  vector < int > di [ MAX ];
  vector < int > mu [ MAX ];
  memset ( dp, 0, sizeof dp );
  memset ( di, 0, sizeof di );
  memset ( mu, 0, sizeof mu );
  for ( int i = 1;
  i <= m;
  i ++ ) {
    for ( int j = 2 * i;
    j <= m;
    j += i ) {
      di [ j ] . push_back ( i );
      mu [ i ] . push_back ( j );
    }
    di [ i ] . push_back ( i );
  }
  for ( int i = 1;
  i <= m;
  i ++ ) dp [ 1 ] [ i ] = 1;
  for ( int i = 2;
  i <= n;
  i ++ ) {
    for ( int j = 1;
    j <= m;
    j ++ ) {
      dp [ i ] [ j ] = 0;
      for ( auto x : di [ j ] ) dp [ i ] [ j ] += dp [ i - 1 ] [ x ];
      for ( auto x : mu [ j ] ) dp [ i ] [ j ] += dp [ i - 1 ] [ x ];
    }
  }
  int ans = 0;
  for ( int i = 1;
  i <= m;
  i ++ ) {
    ans += dp [ n ] [ i ];
    di [ i ] . clear ( );
    mu [ i ] . clear ( );
  }
  return ans;
}

File: TransCoder_tokenized_test_set_functions/python/COUNT_ARRAYS_ADJACENT_ELEMENTS_ONE_DIVIDE_ANOTHER.py
def numofArray ( n, m ) :
    dp = [ [ 0 for i in range ( MAX ) ] for j in range ( MAX ) ]
    di = [ [ ] for i in range ( MAX ) ]
    mu = [ [ ] for i in range ( MAX ) ]
    for i in range ( 1, m + 1 ) :
        for j in range ( 2 * i, m + 1, i ) :
            di [ j ] . append ( i )
            mu [ i ] . append ( j )
        di [ i ] . append ( i )
    for i in range ( 1, m + 1 ) :
        dp [ 1 ] [ i ] = 1
    for i in range ( 2, n + 1 ) :
        for j in range ( 1, m + 1 ) :
            dp [ i ] [ j ] = 0
            for x in di [ j ] :
                dp [ i ] [ j ] += dp [ i - 1 ] [ x ]
            for x in mu [ j ] :
                dp [ i ] [ j ] += dp [ i - 1 ] [ x ]
    ans = 0
    for i in range ( 1, m + 1 ) :
        ans += dp [ n ] [ i ]
        di [ i ] . clear ( )
        mu [ i ] . clear ( )
    return ans
